### PR TITLE
support of environment variable KUBECACHEDIR #500

### PR DESCRIPTION
**Pull request by jjournet**
Fix to support environment variable KUBECACHEDIR, already supported by kubectl, to change the location of folder `~/.kube/cache`.
This is necessary to support having different environments using different Azure identities.

### CODE REVIEW

**Comment by jjournet** [comment]
fix #500

File: pkg/internal/token/options.go
Comment (weinong) [inline pkg/internal/token/options.go:70]: you can just check if `env_TokenCacheDir` is empty or not right after L66 and set it accordingly.

Comment (jjournet) [reply pkg/internal/token/options.go]: does the change follow what you had in mind @weinong ?

if yes, I'll resolver the conversation.
thank you for your advice

File: pkg/internal/token/options.go
Comment (weinong) [inline pkg/internal/token/options.go:66]: let's not change the global variable. how about this?

```go
func NewOptions() Options {
	tokenCacheDir := DefaultTokenCacheDir
	if envTokenCacheDir := os.Getenv("KUBECACHEDIR"); envTokenCacheDir != "" {
		tokenCacheDir = envTokenCacheDir
	}
	return Options{
		LoginMethod:   DeviceCodeLogin,
		Environment:   defaultEnvironmentName,
		TokenCacheDir: tokenCacheDir,
	}
}
```

Comment (jjournet) [reply pkg/internal/token/options.go]: @weinong that's how I implemented it my first commit.

The difference is the use of the inline function to avoid adding a new variable, and I find it more elegant.

I pushed a revert

**Review by weinong — APPROVED** [review]
LGTM

**Comment by jjournet** [comment]
@weinong thank you for the approval

lint didn't like my variable, I changed it, waiting for your approval to run the actions again (sorry for the mistake)

**Pull request merged**
